context/app/static/js/shared-styles/accordions/DetailsAccordion/DetailsAccordion.tsx

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import { AccordionProps } from '@mui/material/Accordion';
+import { AccordionSummaryProps } from '@mui/material/AccordionSummary';
+import { AccordionDetailsProps } from '@mui/material/AccordionDetails';
+import { Typography } from '@mui/material';
+import { Accordion, AccordionDetails, AccordionSummary } from './style';
+
+type DetailAccordionProps = React.PropsWithChildren<{
+  summary: React.ReactNode;
+  summaryProps?: AccordionSummaryProps;
+  detailsProps?: AccordionDetailsProps;
+}> &
+  AccordionProps;
+
+function DetailAccordion({ summary, children, summaryProps, detailsProps, ...accordionProps }: DetailAccordionProps) {
+  // wrap summary in Typography if it's a string to apply basic styling
+  const wrappedSummary =
+    typeof summary === 'string' ? (
+      <Typography variant="body2" component="span" display="inline-block">
+        {summary}
+      </Typography>
+    ) : (
+      summary
+    );
```

Comment:
Pull into a function?

Reply:
I pulled this out into a helper component in the same file and added some basic documentation 👍🏻